README.md

```diff
@@ -1,18 +1,308 @@
-# gulp-build-framework
+# build-framework
 
 ## Options
 
 All build framework options can be set in local /projectConfig.js (<5.x /projectConfig.json)
 
 Description will follow soon, see default config: <https://github.com/frontend-framework/build-framework/blob/master/config.js> 
 
-
 ## Migration checklists
 
 * <https://github.com/frontend-framework/frontend-framework/wiki/Migration-Checklist-3.x-to-4.x>
 * <https://github.com/frontend-framework/frontend-framework/wiki/Migration-Checklist-4.x-to-5.x>
 
-### Troubleshooting
+## Config.js defaults
+
+config.js is the default configuration for all gulp tasks. The following is an overview of the default settings used in this framework.
+
+All configurations can be overwritten via a projectConfig.js inside the root folder of the project using this build-framework.  
+
+### framework internals
+
+###### global
+```
+{
+   <String> src: src, //base folder for all the project files, defaults to 'src'
+   <String> dev: dev, //temporary folder for all files that get created in the serve/build process, defaults to '.tmp'
+   <String> dist: dist, //build folder for the finalized files when running a build process, defaults to 'dist'
+   <String> node: node, //node_modules base folder, defaults to 'node_modules'
+   <String> cwd: cwd, //process.cwd(), current working dir of node project ###TODO should maybe be private###
+   <boolean> isWin: isWin, //checks the os and sets the variable to true if running on a windows system
+   <boolean> debug: false, //toggle debbuging prompts from the tasks (static:hb, browserConfig, partialHelper), default is false
+   <String> dataObject: 'data' //name of the global data object, which can be used in static handlebars templates via {{data.key}}
+   Array<String> resources: ['/resources'], //resource folders inside the src folder, needed to run multiple tasks, can contain multiple different folders
+   Array<String> components: ['/components'] //component folders corresponding to the resource folders   
+   <String> handlebarsHelper: '/js/handlebars.helper.js', //path to dynamic handlebar helpers relative to the resources folder
+   <Object> tasks: {}, //see tasks
+   <Object> externalResources: {} //see external resources   
+   <Object> bowerResources: {} //see bower resources
+}
+```
+###### tasks
+You can toggle certain task in the build-framework. Every Task can be configurated with a boolean. 
+
+This list shows the configurable tasks as well as a brief description about what they do.
+```
+{
```

Comment:
I think this list is not complete, see https://github.com/frontend-framework/build-framework/blob/lazy-load-plugins/config.js#L49 for a more complete task list

Reply:
I added the missing tasks.